### PR TITLE
auto-generate calibration

### PR DESCRIPTION
**Pull request by JonasVautherin**
Tested to work with a Pixhawk 4 and MAVSDK-Python.

### CODE REVIEW

File: src/plugins/calibration/calibration_impl.cpp [hunk 1]
```diff
@@ -220,21 +217,19 @@ void CalibrationImpl::command_result_callback(
         case MAVLinkCommands::Result::TIMEOUT: {
             // Report all error cases.
             const auto timeout_result = calibration_result_from_command_result(command_result);
-            call_user_callback(
-                _calibration_callback,
-                timeout_result,
-                Calibration::ProgressData(false, NAN, false, ""));
+            call_user_callback(_calibration_callback, timeout_result, Calibration::ProgressData());
```
Comment (julianoes) [inline src/plugins/calibration/calibration_impl.cpp:220]:
```suggestion
            call_user_callback(_calibration_callback, timeout_result, Calibration::ProgressData{});
```

File: src/plugins/calibration/include/plugins/calibration/calibration.h [hunk 2]
```diff
@@ -35,99 +43,122 @@ class Calibration : public PluginBase {
...
-        NO_SYSTEM,
-        CONNECTION_ERROR,
-        BUSY,
-        COMMAND_DENIED,
-        TIMEOUT,
-        CANCELLED
+        Unknown, /**< @brief Unknown error. */
+        Success, /**< @brief The calibration process succeeded. */
+        InProgress, /**< @brief Intermediate message showing progress of the calibration process. */
+        Instruction, /**< @brief Intermediate message giving instructions on the next steps required
```
Comment (julianoes) [inline src/plugins/calibration/include/plugins/calibration/calibration.h:52]: I guess this is still to change?

File: src/plugins/calibration/include/plugins/calibration/calibration.h [hunk 3]
```diff
@@ -35,99 +43,122 @@ class Calibration : public PluginBase {
...
-            const float _progress,
-            const bool _has_status_text,
-            const std::string& _status_text) :
-            has_progress(_has_progress),
-            progress(_progress),
-            has_status_text(_has_status_text),
-            status_text(_status_text)
-        {}
+        bool has_progress{
+            false}; /**< @brief Whether this ProgressData contains a 'progress' status or not */
```
Comment (julianoes) [inline src/plugins/calibration/include/plugins/calibration/calibration.h:77]: With C++17 we could make this `std::optional<float> progress`.

Comment (JonasVautherin) [reply src/plugins/calibration/include/plugins/calibration/calibration.h:77]: Sure

File: src/plugins/calibration/include/plugins/calibration/calibration.h (hunk 2 above)
Comment (JonasVautherin) [reply src/plugins/calibration/include/plugins/calibration/calibration.h:52]: Yes, next PR :+1:.

**Pull request merged**